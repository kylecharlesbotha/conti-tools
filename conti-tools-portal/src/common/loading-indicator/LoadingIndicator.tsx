import { Box, CircularProgress, Theme, Typography } from '@mui/material';
import React from 'react';

type LoadingIndicatorProps = {
  text?: string;
  size?: number;
  dense?: boolean;
};

export const LoadingIndicator = ({
  text = '',
  size = 40,
  dense = false
}: LoadingIndicatorProps) => {
  const classes = {
    fillSpace: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center'
    },
    spinner: {
      margin: (theme: Theme) => theme.spacing(2)
    },
    smallSpinner: {
      margin: (theme: Theme) => theme.spacing(0.5)
    }
  } as const;

  return (
    <Box sx={classes.fillSpace}>
      <CircularProgress
        sx={dense ? classes.smallSpinner : classes.spinner}
        size={size}
      />
      {text !== '' && (
        <Typography variant="h5" gutterBottom>
          {text}
        </Typography>
      )}
    </Box>
  );
};
