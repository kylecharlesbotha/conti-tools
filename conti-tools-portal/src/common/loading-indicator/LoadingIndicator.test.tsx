import { render } from '@testing-library/react';
import { LoadingIndicator } from './LoadingIndicator';

describe('Testing loading indicator', () => {
  test('no parameters should use defaults', () => {
    const { baseElement } = render(<LoadingIndicator />);
    expect(baseElement).toMatchSnapshot();
  });

  test('With parameters disclosed should render correctly', () => {
    const { baseElement } = render(
      <LoadingIndicator size={20} dense text="Smaller Loading Indicator" />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
