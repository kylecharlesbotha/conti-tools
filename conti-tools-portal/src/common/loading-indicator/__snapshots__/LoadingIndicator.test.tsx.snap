// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing loading indicator With parameters disclosed should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-wkpw2c"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-17kqp8b-MuiCircularProgress-root"
        role="progressbar"
        style="width: 20px; height: 20px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Smaller Loading Indicator
      </h5>
    </div>
  </div>
</body>
`;

exports[`Testing loading indicator no parameters should use defaults 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-wkpw2c"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-gthy77-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</body>
`;
